lib/fastly_nsq.rb

```diff
@@ -10,15 +10,51 @@
 require 'fastly_nsq/version'
 
 module FastlyNsq
-  def self.logger=(logger)
+  module_function
+
+  def channel=(channel)
+    @channel ||= channel
+  end
+
+  def logger=(logger)
     strategy.logger = logger
   end
 
-  def self.logger
+  def self.preprocessor=(preprocessor)
+    @preprocessor ||= preprocessor
+  end
```

Comment:
Preprocessor hook was intended for common, pre-processor work that applies to all received messages; iirc, the preprocessor hook is mostly used for logging/statsd. If we're moving to daemons, I assume we could get rid of the rake task and preprocessor, since we can expect the upstream client to define *all* relevant processing.